mlxtend/feature_selection/column_selector.py

```diff
@@ -83,7 +83,7 @@ def transform(self, X, y=None):
             types = {type(i) for i in self.cols}
             if len(types) > 1:
                 raise ValueError('Elements in `cols` should be all of the same data type.')
-            if isinstance(self.cols[0], int):
+            elif isinstance(self.cols[0], int):
```

Comment:
I don't think this should be an elif here because we want to check for both things

- first that the column names have the same type
- if the above is true, the second check will check whether these are integer column types.